tests/CookieUtilityTest.php

```diff
@@ -19,31 +20,11 @@
  */
 final class CookieUtilityTest extends TestCase
 {
-    public function testNormalizeDomain(): void
+	/** @dataProvider \DataProvider::domain() */
+    public function testNormalizeDomain(string $input, ?string $expected): void
     {
-        $domain1 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('www.google.de');
-        $domain2 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('www.google');
-        $domain3 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('google.de');
-        $domain4 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('domain-@test.@de');
-        $domain5 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('xn--fsqu00a.xn--0zwm56d');
-        $domain6 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('..');
-        $domain7 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('.');
-        $domain8 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('-_-');
-        $domain9 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('---');
-        $domain10 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('');
-        $domain11 = marcoschwepp\Cookie\CookieUtility::normalizeDomain('www.test.de.');
+		$normalizedDomain = marcoschwepp\Cookie\CookieUtility::normalizeDomain($input);
```

Comment:
Removed all and wrapped up in a DataProvider :) Is more readable and should work much better.